lib/ruby_lsp/requests/support/rubocop_diagnostic.rb

```diff
@@ -80,9 +80,17 @@ def severity
           RUBOCOP_TO_LSP_SEVERITY[@offense.severity.name]
         end
 
-        sig { returns(T.nilable(Interface::CodeDescription)) }
-        def code_description
-          doc_url = RuboCopRunner.find_cop_by_name(@offense.cop_name)&.documentation_url
+        sig { params(config: RuboCop::Config).returns(T.nilable(Interface::CodeDescription)) }
+        def code_description(config)
```

Comment:
What do you think about overriding this method for older versions to avoid the check on every diagnostic? Something like this

```ruby
def code_description(config)
  cop = RuboCopRunner.find_cop_by_name(@offense.cop_name)
  return unless cop

  doc_url = cop.documentation_url
  Interface::CodeDescription.new(href: doc_url) if doc_url
end

begin
  gem("rubocop", ">= 1.64.0")

  def code_description(config)
    cop = RuboCopRunner.find_cop_by_name(@offense.cop_name)
    return unless cop

    doc_url = cop.documentation_url(config)
    Interface::CodeDescription.new(href: doc_url) if doc_url
  end
rescue LoadError
  # do nothing
end
```

Reply:
All these extension will have something returned, the department (`Rails`, `Minitest`, etc.) need to have `DocumentationBaseUrl` set. This is also the reason why you need to pass in the config. All (or at least most of) the official extensions do this properly: https://github.com/search?q=org%3Arubocop+DocumentationBaseUrl+-repo%3Arubocop%2Frubocop+-repo%3Arubocop%2Fdocs.rubocop.org&type=code. `rubocop-sorbet` doesn't set this which is why the expectation diff is so small.

In the end you land in this implementation: https://github.com/rubocop/rubocop/blob/54421d6df7fb245f836fe8074690171c02b3d7b6/lib/rubocop/cop/documentation.rb
Previously this only worked for buildin cops, which I changed. As you noticed asking others to add support explicitly didn't work quite out so well and adding generic support like that in rubocop was just easier than changing all the other extensions.